From 7b827f7e38b86e210140dd5745ec9ab9cb9ad421 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9rgio=20M=2E=20Basto?= <sergio@serjux.com>
Date: Thu, 27 Nov 2025 20:32:54 +0000
Subject: [PATCH 7/9] Fix interface conflicts for Java 25

- Resolve compilation errors caused by both org.gudy.bouncycastle.asn1.DEREncodable
  and java.security.DEREncodable being present in the same scope.

- Adjust EC/ElGamal key interfaces to fix getParams() return type clashes,
  keeping EC and ElGamal parameter types separate.
---
 .../org/gudy/bouncycastle/jce/interfaces/ECPrivateKey.java   | 5 +++++
 .../org/gudy/bouncycastle/jce/interfaces/ECPublicKey.java    | 5 +++++
 .../gudy/bouncycastle/jce/interfaces/ElGamalPrivateKey.java  | 3 +++
 .../gudy/bouncycastle/jce/interfaces/ElGamalPublicKey.java   | 3 +++
 4 files changed, 16 insertions(+)

diff --git a/core/src/org/gudy/bouncycastle/jce/interfaces/ECPrivateKey.java b/core/src/org/gudy/bouncycastle/jce/interfaces/ECPrivateKey.java
index 2224d9f21..30db44109 100644
--- a/core/src/org/gudy/bouncycastle/jce/interfaces/ECPrivateKey.java
+++ b/core/src/org/gudy/bouncycastle/jce/interfaces/ECPrivateKey.java
@@ -2,6 +2,7 @@ package org.gudy.bouncycastle.jce.interfaces;
 
 import java.math.BigInteger;
 import java.security.PrivateKey;
+import org.gudy.bouncycastle.jce.spec.ECParameterSpec;
 
 /**
  * interface for Elliptic Curve Private keys.
@@ -13,4 +14,8 @@ public interface ECPrivateKey
      * return the private value D.
      */
     public BigInteger getD();
+    // Resolve conflict between ECKey and potential defaults
+		@Override
+		ECParameterSpec getParams();
+
 }
diff --git a/core/src/org/gudy/bouncycastle/jce/interfaces/ECPublicKey.java b/core/src/org/gudy/bouncycastle/jce/interfaces/ECPublicKey.java
index 58629e44c..b87c274cd 100644
--- a/core/src/org/gudy/bouncycastle/jce/interfaces/ECPublicKey.java
+++ b/core/src/org/gudy/bouncycastle/jce/interfaces/ECPublicKey.java
@@ -3,6 +3,7 @@ package org.gudy.bouncycastle.jce.interfaces;
 import java.security.PublicKey;
 
 import org.gudy.bouncycastle.math.ec.ECPoint;
+import org.gudy.bouncycastle.jce.spec.ECParameterSpec;
 
 /**
  * interface for elliptic curve public keys.
@@ -14,4 +15,8 @@ public interface ECPublicKey
      * return the public point Q
      */
     public ECPoint getQ();
+
+		// ElGamalPrivateKey.java
+		@Override
+		ECParameterSpec getParams();
 }
diff --git a/core/src/org/gudy/bouncycastle/jce/interfaces/ElGamalPrivateKey.java b/core/src/org/gudy/bouncycastle/jce/interfaces/ElGamalPrivateKey.java
index 4a42e4d0e..0472f4d5f 100644
--- a/core/src/org/gudy/bouncycastle/jce/interfaces/ElGamalPrivateKey.java
+++ b/core/src/org/gudy/bouncycastle/jce/interfaces/ElGamalPrivateKey.java
@@ -2,9 +2,12 @@ package org.gudy.bouncycastle.jce.interfaces;
 
 import java.math.BigInteger;
 import java.security.PrivateKey;
+import org.gudy.bouncycastle.jce.spec.ElGamalParameterSpec;
 
 public interface ElGamalPrivateKey
     extends ElGamalKey, PrivateKey
 {
     public BigInteger getX();
+		@Override
+		ElGamalParameterSpec getParams();
 }
diff --git a/core/src/org/gudy/bouncycastle/jce/interfaces/ElGamalPublicKey.java b/core/src/org/gudy/bouncycastle/jce/interfaces/ElGamalPublicKey.java
index 055dfc89d..fd9ea14be 100644
--- a/core/src/org/gudy/bouncycastle/jce/interfaces/ElGamalPublicKey.java
+++ b/core/src/org/gudy/bouncycastle/jce/interfaces/ElGamalPublicKey.java
@@ -2,9 +2,12 @@ package org.gudy.bouncycastle.jce.interfaces;
 
 import java.math.BigInteger;
 import java.security.PublicKey;
+import org.gudy.bouncycastle.jce.spec.ElGamalParameterSpec;
 
 public interface ElGamalPublicKey
     extends ElGamalKey, PublicKey
 {
     public BigInteger getY();
+		@Override
+		ElGamalParameterSpec getParams();
 }
-- 
2.51.1

